src/nativeapm/nativeapm.ts

```diff
@@ -0,0 +1,469 @@
+     */
+    theme: NativeApmThemeConfig;
+
+    /**
+     * Prefilled data of the user
+     * @type {PrefilledData}
+     */
+    prefilledData: PrefilledData = {};
+
+    /**
+     * Count of retries for capturing payment
+     * @type {number}
+     */
+    captureRetries = 0;
+
+    /**
+     * NativeAPM constructor
+     * @param  {NativeApmConfig} config
+     */
+    constructor(
+      processOutInstance: ProcessOut,
+      paymentConfig: NativeApmConfigType
+    ) {
+      this.processOutInstance = processOutInstance;
+      this.paymentConfig = new NativeApmPaymentConfig(paymentConfig);
+      this.theme = new NativeApmThemeConfig();
+    }
+
+    /**
+     * This function mounts the Native APM widget on the page
+     * @param  {string} containerSelector
+     */
+    public mount(containerSelector: string) {
+      this.napmContainer = document.querySelector(containerSelector);
+
+      if (!this.napmContainer) {
+        throw new Exception(
+          'default',
+          'Element with this selector does not exist. You must provide valid element selector'
+        );
+      }
+
+      this.loadView(new NativeApmSpinner(this.theme).getSpinnerElement());
+
+      this.getGatewayConfiguration({
+        onFetch: EventsUtils.dispatchWidgetLoadingEvent,
+        onSuccess: this.onGatewayConfigurationSuccess.bind(this),
+        onError: this.onGatewayConfigurationError.bind(this),
+      });
+    }
+
+    /**
+     * This function gets gateway configuration of the Native APM widget
+     */
+    private getGatewayConfiguration(actions: {
+      onFetch: Function;
+      onSuccess: Function;
+      onError: Function;
+    }) {
+      const paymentConfig = this.paymentConfig.getConfig();
+
+      actions.onFetch();
+
+      this.processOutInstance.apiRequest(
+        'GET',
+        `invoices/${paymentConfig.invoiceId}/native-payment/${paymentConfig.gatewayConfigurationId}`,
+        {},
+        actions.onSuccess.bind(this),
+        actions.onError.bind(this)
+      );
+    }
+
+    private onGatewayConfigurationSuccess(data: GatewayConfiguration) {
+      if (!data.success) {
+        const errorView = new NativeApmErrorView({}, this.theme);
+
+        EventsUtils.dispatchGatewayConfigurationErrorEvent(data);
+
+        return this.loadView(errorView.getViewElement());
+      }
+
+      this.gatewayConfiguration = data;
+
+      if (
+        this.gatewayConfiguration.native_apm &&
+        this.gatewayConfiguration.native_apm.parameters.length
+      ) {
+        const nativeApmFormView = new NativeApmFormView(
+          this.gatewayConfiguration.native_apm,
+          this.proceedPayment.bind(this),
+          this.theme,
+          this.prefilledData
+        );
+
+        EventsUtils.dispatchWidgetReadyEvent();
+
+        return this.loadView(nativeApmFormView.getViewElement());
+      }
+
+      if (
+        this.gatewayConfiguration.native_apm &&
+        this.gatewayConfiguration.native_apm.parameters.length === 0
```

Comment:
This is a weird case, but it happened to me when I was testing it, so I though that it might be a good idea to have some logic which will cover that